myia/parser_opt.py

```diff
@@ -0,0 +1,90 @@
+"""Parser optimizations, applicable on graph after parsing."""
+from collections import deque
+
+from myia.basics import (
+    global_universe_getitem,
+    global_universe_setitem,
+    make_handle,
+)
+from myia.ir.graph_utils import get_node_users, toposort
+from myia.ir.node import SEQ, Apply, Graph
+
+
+def _replace_apply_node(root_graph: Graph, node: Apply, new_node):
+    """Replace node by new_node in given graph.
+
+    :param root_graph: graph in which to make replacements.
+        Replacement will be recursively applied to graph and all children closures.
+    :param node: node to replace
+    :param new_node: node to replace with. May be None to delete node.
+    """
+    mapping = {node: new_node}
+    mapping_seq = {node: node.edges[SEQ].node} if SEQ in node.edges else {}
```

Comment:
If you want to delete the SEQ edge, map it to `None`.  An empty dictionary will just use the normal mapping